bin/utest/sigaction.c

```diff
@@ -45,3 +53,47 @@ int test_sigaction_handler_returns(void) {
 
   return 0;
 }
+
+#define NCHLD 7
+
+static volatile int nreaped;
+
+static void sigchld_handler(int signo, siginfo_t *si, void *uctx) {
+  assert(si->si_code == CLD_EXITED);
```

Comment:
This is not the only possibility. What if the child was _continued_ or _stopped_ or _killed_? Could you as well check `si_pid` and `si_status`.